internal/service/databaseuserapi/resource.go

```diff
@@ -66,13 +70,13 @@ func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resou
 
 	respBody, err := io.ReadAll(apiResp.Body)
```

Comment:
```suggestion
	defer apiResp.Body.Close()
	respBody, err := io.ReadAll(apiResp.Body)
```

we need to do this before each call to `io.ReadAll`, feel free to do it in this PR or a follow-up PR, e.g. as in the SDK [here](https://github.com/mongodb/atlas-sdk-go/blob/2c1a8a0a1658ebb7bd8ef21a3b62cfd59658563d/admin/api_database_users.go#L247)

Reply:
thanks, following example from [go docs](https://pkg.go.dev/net/http#example-Get) calling close directly after the `ReadAll`